### ISSUE TITLE
Tray destroy method crashes the app

### ISSUE THREAD

**Issue by antonio-ramadas** [electron/electron#9982]
* Electron version: 1.6.11
* Operating system: macOS 10.12.5

### Expected behaviour

Create and destroy the tray instance multiple times with no issues or faulty behaviour.

### Actual behavior

Create the tray instance and destroy it, one after the other. You may need to repeat these steps multiple times. The crash will eventually happen when destroying the tray instance. The app crash seems to happen randomly. From my experience, it may happen on the first time or only after multiple repetitions (7 or so). The app will crash with `Received signal 11 SEGV_MAPERR` followed by the stack dump.

### How to reproduce

I've created a repository to reproduce the issue. There's only one branch and the instructions, as some other information, is in the README.md file at the root level. The tray instance has a context menu, built from the template, and the window has a button to create the instance. Lastly, the repository is based on the sample app provided by Electron.

Repository: https://github.com/antonio-ramadas/electron-tray-app-crash

**Comment by microshine** [comment]
I've got the same problem 

Electron: v1.7.6
OS: OS X 10.13.1

**Comment by matt-way** [comment]
Not sure if exactly related, but I found that destroying the tray would crash on my machine due to a mouse tracking race condition.

See: https://github.com/electron/electron/pull/11272

**Comment by Splode** [comment]
I'm encountering this issue as well.

Electron `v.1.7.5`
OS: OS X `10.13.3`

**Comment by Negan1911** [comment]
I think that i've got it, just wrap the `.destroy()` in a `setTimeout()` like this:
```js
setTimeout(() => {
  myTray.destroy()
}, 0)
```

**Comment by LabDeve** [comment]
I've got the same problem...

Electron v.2.0.3
OS: OS X 10.13.5

**Comment by shameister** [comment]
Subscribing this, because we got the same issue here.

**Comment by miniak** [comment]
Seems like a duplicate of https://github.com/electron/electron/issues/14036
